### PR TITLE
step3

### PR DESCRIPTION
**Pull request by EileenWang-10001010**
## What
continue to step 3.9
In @app.get("/search") we can use keyword to search name, ex: 'jacket', or search category, ex: 'fashion'.
Others is the same as the requirement.

## CHECKS :warning:

Please make sure you are aware of the following.

- [ ] **The changes in this PR doesn't have private information

### CODE REVIEW

File: python/main.py [hunk 1]
```diff
@@ -18,14 +20,98 @@
...
+    
+    conn.close()
+    return {f"items:{r}"}
+
+@app.get("/search")
+async def search_items(keyword: str):
+    
+    conn = sqlite3.connect('mercari.sqlite3')
+    c = conn.cursor()
+    c.execute("""SELECT * from items WHERE name == (?)""",[keyword])
```
Comment (yuting0203) [inline python/main.py]: It seems that some items containing the keyword will not be return by using '=='. E.g., When searching "jacket", the item "kids jacket" will not show.

The "LIKE" operator may be useful in this case.
[https://www.sqlitetutorial.net/sqlite-like/](https://www.sqlitetutorial.net/sqlite-like/)

File: python/main.py [hunk 2]
```diff
@@ -18,14 +20,98 @@
...
+    searchByName = c.fetchall()
+    if (len(searchByName)!=0):
+        r =[]
+        for item in searchByName:
+            c.execute("""SELECT * from category WHERE id == (?)""",[item[2]])
+            r.append({f"name:{item[1]}, category:{(c.fetchone())[1]}, image:{item[3]}"})
```
Comment (yuting0203) [inline python/main.py]: Maybe you can use INNER JOIN when returning responses.
FYI. [https://www.techonthenet.com/sqlite/joins.php](https://www.techonthenet.com/sqlite/joins.php)

**Review by yuting0203 — COMMENTED** [review]
I left some comments. Hope the comments are useful for you.

File: python/main.py (hunk 1 above)
Comment (EileenWang-10001010) [reply python/main.py]: Thank you~
I' ll change it~

File: python/main.py (hunk 2 above)
Comment (EileenWang-10001010) [reply python/main.py]: This may be faster and more explainable than my previous attempt, thank you so much~

**Pull request merged**